### DISCUSSION TITLE
Transaction commit not working for sub functions

### DISCUSSION THREAD

**Discussion by ankurgoels** [laravel/framework#34637] [Q&A]
- Laravel Version: ~6.0.4~ 6.18.41
- PHP Version: 7.2.31
- Database Driver & Version: Mysql 5.7

### Description:
A very weird issue is happening. I have a job running using supervisor with multiple workers. The jobs call a sub-function that creates a model and returns it. Other DB actions are being performed as well but sometimes the model which is created in sub-function is not written to DB. Other DB operations that are using the ID from the model and are being populated well, but the actual model entry is missing from DB.

### Steps To Reproduce:
Code
```
class Main{
    mainFunction() {
        try {
            DB::beginTransaction();

            $data['name'] = "name";
            $result = (new anotherClass($data))->create();

            // result primary key in other model

            DB::commit();
        } catch(Exception $e) {
            DB::rollBack();
        }
    }
}

class anotherClass() {
    // skipping construct

    public function create() {
        $model = Model::create([
            'name' => $this->data['name']
        ]);
        return $model;
    }
}
```

entry created in anotherClass->create() is not written to DB but other models are written and contains the id from anotherClass->create(). This does not happen every time but sometimes it does. All tables types are *InnoDB*

**Comment by driesvints** [comment]
Can you update to the latest 6.x version and see if the problem persists?

**Comment by ankurgoels** [comment]
@driesvints Have updated the updated to 6.18.41. But still, the issue persists.

Also, I am using 2 DB connections in the job could this cause this behavior?

**Comment by mfn** [comment]
Do you have the same problem when using `DB::transaction()`?

**Reply by ankurgoels** [reply]
yes, `DB::transaction()` doesn't solves the problem

**Comment by algermakiputin** [comment]
Same issue here. Has anyone come up with a fix?

**Reply by ankurgoels** [reply]
I was able to debug... Mine was due to a server memory issue, the server got out of memory and the changes were not getting committed. Check if this helps.

**Reply by rifat-simoom** [reply]
Hello ankurgoels, I am having the same problem. Do you know any steps to generate the problem?

**Comment by agounane** [comment]
i had the same issue, when the database table name is defined on the model (when the database table name is not default as Laravel expect)
